src/commands/xmentor.ts

```diff
@@ -1,60 +1,89 @@
-import { GluegunToolbox } from 'gluegun'
+import { GluegunToolbox } from 'gluegun';
+
+const LANGUAGE_TEST_FILE_MAPPER: { [k: string]: string } = {
+  'typescript': 'test.ts',
+  'javascript': 'spec.js',
+};
+
+const LANGUAGE_RUN_COMMAND_MAPPER: { [k: string]: string } = {
+  'typescript': `npx tsc --noEmit -p . && npx jest --i --no-cache`,
+  'javascript': 'jest --no-cache ./*',
+};
 
 export default {
   name: 'xmentor',
   run: async (toolbox: GluegunToolbox) => {
-    const { print, parameters, system, prompt, filesystem } = toolbox
+    const { print, parameters, system, prompt, filesystem } = toolbox;
 
-    const exerciseUUID = parameters.string
+    const exerciseUUID = parameters.string;
+    const spinner = print.spin({ text: 'Downloading student solution.', spinner: 'dots' });
 
-    const spinner = print.spin({ text: 'Downloading student solution.', spinner: 'dots' })
+    if (!exerciseUUID) return spinner.fail('Please provide an UUID');
```

Comment:
It's not a bad idea but ultimately is not necessary because the call to exercism CLI download command on line 24 will deal with cases of empty or bad UUID and throw a well crafted error message that we are already catching on line 62 and displaying to the user on line 63. So you can probably do away with this.